src/Category/Algebra.agda

```diff
@@ -0,0 +1,119 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Algebras for endofunctors
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Category.Category using (Category)
+
+module Category.Algebra {ℓ} (cat : Category ℓ) where
+
+open import Level
+open Category cat
+open import Category.Functor cat cat
+  renaming (RawFunctor to RawEndoFunctor)
+
+module _ {fun : Obj → Obj} (F : RawEndoFunctor fun) where
+  open RawEndoFunctor F
+  
+  record Algebra : Set (suc ℓ) where
+    field
+      Carrier  : Obj
+      evaluate : fun Carrier ⇒ Carrier
+  open Algebra public
+
+  open import Category.Structures
+  module AlgebraCategory (isCat : IsCategory cat) (isFun : IsFunctor cat cat F) where
+    open IsCategory isCat
+    open IsFunctor isFun
+    open ≈-Reasoning
+
+    record AlgebraMorphism (algu : Algebra) (algv : Algebra) : Set ℓ where
+      field
+        translation : Carrier algu ⇒ Carrier algv
+        commutes    : evaluate algv ∘ fmap translation ≈ translation ∘ evaluate algu
+    open AlgebraMorphism
+    
+    idAlgebraMorphism : ∀ (alg : Algebra) → AlgebraMorphism alg alg
+    idAlgebraMorphism alg = record
+      { translation = id
+      ; commutes = begin
+        evaluate alg ∘ fmap id ≈⟨ fmap-identity over evaluate alg ⟩
+        evaluate alg ∘ id      ≈⟨ identityʳ ⟩
+        evaluate alg           ≈⟨ ≈sym identityˡ ⟩
+        id ∘ evaluate alg      ∎
+      }
+
+    infixr 9 _∘alg_
+    _∘alg_ : ∀ {algu algv algw : Algebra}
+           → AlgebraMorphism algv algw → AlgebraMorphism algu algv → AlgebraMorphism algu algw
+    _∘alg_ {algu = au} {algv = av} {algw = aw} vw uv = record
+      { translation = tvw ∘ tuv
+      ; commutes = begin
+        algw ∘ fmap (tvw ∘ tuv)      ≈⟨ fmap-homomorphism over algw ⟩
+        algw ∘ (fmap tvw ∘ fmap tuv) ≈⟨ ≈sym assoc ⟩
+        (algw ∘ fmap tvw) ∘ fmap tuv ≈⟨ vw .commutes under fmap tuv ⟩
+        (tvw ∘ algv) ∘ fmap tuv      ≈⟨ assoc ⟩
+        tvw ∘ algv ∘ fmap tuv        ≈⟨ uv .commutes over tvw ⟩
+        tvw ∘ (tuv ∘ algu)           ≈⟨ ≈sym assoc ⟩
+        (tvw ∘ tuv) ∘ algu           ∎
+      } where
+        algu = evaluate au
+        algv = evaluate av
+        algw = evaluate aw
+        tvw = translation vw
+        tuv = translation uv
+
+    module _ {algu : Algebra} {algv : Algebra} where
+      infix 4 _≈alg_
+      record _≈alg_ (m : AlgebraMorphism algu algv) (n : AlgebraMorphism algu algv) : Set ℓ where
+        field
+          morphs-eq : translation m ≈ translation n
+      open import Relation.Binary
```

Comment:
This should go at the top of the file. If the names conflict then they should be qualified by renaming the module as `B` or something similar.